src/libs/registry/docset.cpp

```diff
@@ -674,10 +713,23 @@ static void matchFuzzy(const char *needle, int needleLength,
 
         if (!found) {
             // End of haystack, char not found.
-            *start = -1;
+            if (bestRecScore != -1) {
+                // Can still match with the same constraints if matching started later
+                // (smaller distance from first char to 2nd char)
+                *start = bestRecStart;
+                *length = bestRecLength;
+            } else {
+                *start = -1;
+            }
             return;
         }
     }
+
+    int score = scoreFuzzy(haystack, *start, *length);
```

Comment:
Above only the cases of some `recStart > *start` were considered - here we compare all the later-starting scores against the 'earliest' score. For example, when searching for in "foobaz.foobar" could yield two different scores - matching "[**foo**]baz.foo[**bar**]" or "foobaz.[**foobar**]" - in the latter case of exact match the score would be larger.

I hope to address the other comments later today.